libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Instrumentation/SFMethodInterceptor.m

```diff
@@ -85,9 +85,23 @@ - (instancetype)init {
 }
 
 - (void)dealloc {
-    @synchronized (InterceptorsForClassAndSelector) {
```

Comment:
This was unbalanced.  If the equivalent (duplicate) object had been previously added, this was removing it.  Moved this to balance against the enable/swizzle functionality.